go/arrow/datatype_fixedwidth.go

```diff
@@ -24,6 +24,14 @@ func (t *BooleanType) Name() string { return "bool" }
 // BitWidth returns the number of bits required to store a single element of this data type in memory.
 func (t *BooleanType) BitWidth() int { return 1 }
 
+type FixedSizeBinaryType struct {
```

Comment:
we can't add this type to the `FixedWidthTypes` struct (well, we could, but it wouldn't make much sense.)
but perhaps we could test `FixedSizeBinaryType` implements the `FixedWidthDataType` interface using the usual `_ FixedWidthDataType = (*FixedSizeBinaryType)(nil)` trick?